frontend/packages/api/src/lib/fetcher.ts

```diff
@@ -90,5 +87,23 @@ export const createFetcher = (site: string, baseUrl: string): Client => {
 		})
 	}
 
-	return client
+	return fetcher
+}
+
+function getSearchParams(query?: QueryParams) {
+	if (!query) {
+		return ''
+	}
+
+	// overrides type of init in URLSearchParams constructor because it's not correct and accepts QueryParams
+	const searchParams = new URLSearchParams(
+		Object.entries(query).reduce<Record<string, string>>(
+			(acc, [k, v]) => Object.assign(acc, { [k]: v.toString() }),
```

Comment:
I think we don't need to create new object on every iterations and we can mutate `acc`
```
acc[k] = v.toString();
```